src/gc-pages.c

```diff
@@ -0,0 +1,209 @@
+// This file is a part of Julia. License is MIT: http://julialang.org/license
+
+#include "gc.h"
+#ifndef _OS_WINDOWS_
+#  include <sys/resource.h>
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// A region is contiguous storage for up to REGION_PG_COUNT naturally aligned GC_PAGE_SZ pages
+// It uses a very naive allocator (see jl_gc_alloc_page & jl_gc_free_page)
+#if defined(_P64) && !defined(_COMPILER_MICROSOFT_)
```

Comment:
I think we should be able to remove the msvc ifdef now if it's dynamically allocated